syncer/streamer_controller.go

```diff
@@ -43,39 +43,42 @@ const (
 // For other implementations who implement StreamerProducer and Streamer can easily take place of Syncer.streamProducer
 // For test is easy to mock
 type StreamerProducer interface {
-	generateStreamer(pos mysql.Position) (streamer.Streamer, error)
+	generateStreamer(location binlog.Location) (streamer.Streamer, error)
 }
 
 // Read local relay log
 type localBinlogReader struct {
 	reader *streamer.BinlogReader
 }
 
-func (l *localBinlogReader) generateStreamer(pos mysql.Position) (streamer.Streamer, error) {
-	return l.reader.StartSync(pos)
+func (l *localBinlogReader) generateStreamer(location binlog.Location) (streamer.Streamer, error) {
+	return l.reader.StartSync(location.Position)
 }
 
 // Read remote binlog
 type remoteBinlogReader struct {
 	reader     *replication.BinlogSyncer
 	tctx       *tcontext.Context
+	flavor     string
 	EnableGTID bool
 }
 
-func (r *remoteBinlogReader) generateStreamer(pos mysql.Position) (streamer.Streamer, error) {
+func (r *remoteBinlogReader) generateStreamer(location binlog.Location) (streamer.Streamer, error) {
 	defer func() {
 		lastSlaveConnectionID := r.reader.LastConnectionID()
 		r.tctx.L().Info("last slave connection", zap.Uint32("connection ID", lastSlaveConnectionID))
 	}()
 
-	// FIXME: can enable GTID
 	if r.EnableGTID {
-		// NOTE: our (per-table based) checkpoint does not support GTID yet
-		return nil, terror.ErrSyncerUnitRemoteSteamerWithGTID.Generate()
+		streamer, err := r.reader.StartSyncGTID(location.GTIDSet.Origin())
+		if err != nil {
+			return nil, err
```

Comment:
```suggestion
			return nil, terror.ErrSyncerUnitRemoteSteamerStartSync.Delegate(err)
```

or combine L73 ~ L77 to just two lines as L82 ~ L83.

Reply:
update